FitTracker/app/src/main/java/com/pps/globant/fittracker/mvp/presenter/LoginPresenter.java

```diff
@@ -11,7 +21,44 @@
     public LoginPresenter(LoginModel model, LoginView view) {
         this.model = model;
         this.view = view;
-        view.setLabel("Hi there");
     }
 
+    @Subscribe
+    public void onFetchingFbUserDataCompletedEvent(FacebookLoginProvider.FetchingFbUserDataCompletedEvent event) {
+        Resources res= Objects.requireNonNull(view.getActivity()).getResources();
+        view.setLabelFb(String.format(res.getString(R.string.loged_in_name),event.fbUser.getName()));
+    }
+
+    @Subscribe
+    public void onCantRetrieveAllTheFieldsRequestedsEvent(FacebookLoginProvider.CantRetrieveAllTheFieldsRequestedsEvent event) {
+        Resources res= Objects.requireNonNull(view.getActivity()).getResources();
+        view.setLabelFb(res.getString(R.string.fb_login_error_message));
+        Log.e(res.getString(R.string.facebook),res.getString(R.string.fb_login_error_logPrintErrorCantRetrieveAllTheFields));
+    }
+
+    @Subscribe
+    public void onFetchingFbUserDataErrorEvent(FacebookLoginProvider.FetchingFbUserDataErrorEvent event) {
+        Resources res= Objects.requireNonNull(view.getActivity()).getResources();
+        view.setLabelFb(res.getString(R.string.fb_login_error_message));
+        Log.e(res.getString(R.string.facebook),String.format(res.getString(R.string.fb_login_error_logPrintErrorCantFetchData),event.facebookException.toString()));
+    }
+
+    @Subscribe
+    public void onLogOutCompleteEvent(FacebookLoginProvider.LogOutCompleteEvent event) {
+        Resources res= Objects.requireNonNull(view.getActivity()).getResources();
+        view.setLabelFb(res.getString(R.string.not_loged_in));
+    }
+
+    @Subscribe
+    public void onFbButtonPressedEvent(LoginView.FbButtonPressedEvent event) {
+        model.registerFbCallbacks();
+    }
+
+    public void fbLogOut() {
+        model.fbLogOut();
+    }
+
+    public void redirectActivityResults(int requestCode, int resultCode, Intent data) {
+        CallBackManagerProviderForFb.getCallbackManager().onActivityResult(requestCode, resultCode, data);
```

Comment:
Singleton **CallBAckManagerProviderForFb** not needed, you can send a **CallbackManager** instance in the model constructor and use it here.

```  
callbackManager = CallbackManager.Factory.create();
presenter = new LoginPresenter(
                new LoginModel(
                        callbackManager,
                        BusProvider.getInstance()
                ),
                new LoginView(
                        this,
                        BusProvider.getInstance()
                )
        );
```

Then in the activity **onActivityResult** do:

```
 @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        callbackManager.onActivityResult(requestCode, resultCode, data);
    }
```

Reply:
about the second comment... nacho told me to move the redirection to the callbackmanager of the activity result to the presenter (or that's what i understood), see https://github.com/priettt/FitTracker/pull/2#discussion_r219838068. I already undid this change to independise the presenter from the activity's life cicle, just a comment